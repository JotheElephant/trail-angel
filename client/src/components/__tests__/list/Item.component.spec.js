'use strict';

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Item from '../../list/Item.component';
import homeData from '../../../../__tests__/fixtures/home-data';
import geolocationData from '../../../../__tests__/fixtures/geolocation-data';

jest.mock('../../list/Details.component', () => 'Details');
jest.mock('../../../api/google-api');
jest.mock('../../../api/darksky-api');

test('renders correctly', () => {
  const data = homeData[0];
  const tree = renderer.create(
    <Item favorites={data}
          userLocation={geolocationData}
          {...data}
    />
  );
  expect(tree).toMatchSnapshot();
});