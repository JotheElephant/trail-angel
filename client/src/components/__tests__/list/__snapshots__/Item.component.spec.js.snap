exports[`test renders correctly 1`] = `
<View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }>
      <View
        size={25}
        style={
          Object {
            "alignItems": "center",
            "flex": 25,
            "flexDirection": "column",
            "padding": 10,
          }
        }>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
            }
          }>
          <Image
            source={
              Object {
                "uri": "https://maps.googleapis.com/maps/api/place/photo?maxwidth=100&photoreference=CoQBdwAAAFVOF9vQsY7u9MuWaDAA2VvNr8wDc5lTq99-BHXmizDXvXZfAK49VnQP80jvEDz7nEi5zvnD-ve9vOuUzPLbcs7zsD3xHO74c0bhUuar19_J7dLcu3SUuXuE5YcaVschjUI-5e6NY-fdr_xu_fVF1vVmQtkuU-fDCgyYc-qB0KHYEhC7hA-9Mlehu3OI3VEslNqQGhQqas9upAHkoejM0jSkP7IVW1ONVA&key=AIzaSyDB5zzhpIGeUa2tU4vsqkUAVH0a5q-i_As",
              }
            }
            style={
              Object {
                "borderRadius": 30,
                "height": 60,
                "marginBottom": 10,
                "width": 60,
              }
            } />
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
            }
          }>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ]
            }
            testID={undefined}>
            <View>
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#e56452",
                      "fontSize": 20,
                    },
                    undefined,
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }>
                ï€†
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        size={75}
        style={
          Object {
            "alignItems": "center",
            "flex": 75,
            "flexDirection": "column",
            "marginLeft": 10,
            "marginRight": 10,
            "paddingTop": 10,
          }
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#2f5e4e",
              "fontSize": 15,
              "fontWeight": "500",
              "marginBottom": 5,
              "textAlign": "left",
            }
          }>
          Old Salmon River Trailhead
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#786048",
              "marginBottom": 5,
            }
          }>
          Welches
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#727b24",
            }
          }>
          Rating: 
          4.7
        </Text>
      </View>
      <View
        size={25}
        style={
          Object {
            "alignItems": "center",
            "flex": 25,
            "flexDirection": "column",
            "padding": 10,
          }
        }>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail">
            
          </Text>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
            }
          }>
          <ActivityIndicator
            animating={true}
            color="#5e9fa1"
            hidesWhenStopped={true}
            size="small"
            style={
              Object {
                "opacity": 0.8,
              }
            } />
        </View>
      </View>
    </View>
  </View>
</View>
`;
